### PR TITLE
Mergin Development into Main

### PR DESCRIPTION
**Pull request by ayearwood81**
- for peer review process

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,231 @@
...
-  min-height: 100vh;
-  display: flex;
+form button {
+  width:auto;
+  align-self: flex-end;
+  background-color: rgb(12, 112, 12);
+  font-family: 'Marck Script', cursive;
+  color:  rgba(250, 235, 215, 0.8);
+  border: none;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.397);
```
Comment (colindamelio) [inline src/App.css:75]: great work

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+      <h1>Confidant</h1>
+      <Routes>
+        <Route path ="/" element={
+          <>
+            <EntryForm />
+            <OldEntries />
+          </>
+        }/>;
+        <Route path ="/:entryID" element={<SelectedEntry />}/>;
+        <Route path="*" element={<h2>Error</h2>}/>;
```
Comment (vinni-cmd) [inline src/App.js]: Nice to see routes in action. Looking forward to one day implementing this in my project too.

File: src/EntryForm.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      setBody(e.target.value);
+    }
+    
+    const handleTitleChange = (e) => {
+      setTitle(e.target.value);
+    }
+  
+    const handleSubmit = (e) => {
+      e.preventDefault();
+
```
Comment (vinni-cmd) [inline src/EntryForm.js:28]: Some folks might comment on the spaces here but I think it helps make your code more easy to read

File: src/EntryForm.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      if (body && title) {
```
Comment (vinni-cmd) [inline src/EntryForm.js:29]: your code is nice and clean and easy to follow.

File: src/EntryForm.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        const dbRef = ref(database);
+        console.log(body, title, entryDate);
+  
+        const diaryEntry = {
+          content:body,
+          title:title,
+          date: entryDate,
+        }
+    
+        push(dbRef, diaryEntry);
```
Comment (vinni-cmd) [inline src/EntryForm.js:40]: Using firebase here is amazing. You don't have to pass props around like a crazy person (like I did)!

File: src/OldEntries.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+const OldEntries = () => {
+
+    const [oldEntries, setOldEntries] = useState([]);
+    const inputRef = useRef();
+
+    useEffect( () => {
+        // create variable to hold database details
+        const database = getDatabase(firebase);
+        // create variable that makes a reference to database
```
Comment (vinni-cmd) [inline src/OldEntries.js:15]: nice use of comments

File: src/OldEntries.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            }
+        })
+    }
+    
+    return (
+        <div className="wrapper bottomWrapper">
+            <div className="collapsibleMenu">
+                <input ref={inputRef} type="checkbox" id="menu"></input>
+                <label htmlFor="menu">
+                    <button onClick={handleToggle} className="menuButton"><i className="fa-regular fa-square-caret-down"></i>Previous Entries</button>
```
Comment (vinni-cmd) [inline src/OldEntries.js:85]: maybe in this instance and at a few other locations you can force some of the attributes onto a new indented line just to make it easier to read. I def need to do this too!

File: src/App.css [hunk 8]
```diff
@@ -1,38 +1,231 @@
...
+.collapsibleMenu {
+  z-index: 10;
+  margin-top: 1%;
+  display:flex;
   flex-direction: column;
-  align-items: center;
-  justify-content: center;
-  font-size: calc(10px + 2vmin);
-  color: white;
+  align-items: flex-start;
```
Comment (vinni-cmd) [inline src/App.css]: Is `align-items: flex-start` not the default behaviour?

File: src/App.css [hunk 9]
```diff
@@ -1,38 +1,231 @@
...
+input:checked ~ .menuContent {
+  max-height: 100%;
+}
+
+.selectedWrapper {
+  margin-top: 25px;
+}
+
+.selectedWrapper h2 {
+  font-family: 'Marck Script', cursive;
```
Comment (vinni-cmd) [inline src/App.css]: Not required but if you want to, you can use CSS custom properties. Define them in the root at the top of your style file and then reference them with the var CSS function. 

Example:
```
:root {
  --main-font: 'Marck Script', cursive;
}
```

And then where ever you want to use it do so like this:

```
element {
  font-family: var(--main-font);
}
```

Comment (vinni-cmd) [inline src/App.css]: Have you considered applying the default font for your app to the `body` element? Lots of elements inside the body will then just inherit the font automatically.

**Pull request merged**